zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -132,14 +132,34 @@ pub(crate) fn get_spendable_sapling_notes(
     anchor_height: BlockHeight,
```

Comment:
This should be taking a transaction since it needs to be atomic.

Reply:
Hmmm... that presents a bit of a problem, because it means that anything accessing this needs at least &mut reference to the wallet database in order to construct the transaction, which is a more far-reaching change. I think it's okay for this to be non-atomic, because if the initial filter (on checking whether the subtree containing the anchor height is complete to the anchor height) passes, then the only way that subsequent results could be inconsistent with that would require a rollback, and the error that this would be preventing (a transaction constructed using the results of this query is rejected by the network because it attempts a double-spend) exists regardless.